### PR TITLE
Slack reporter and KonanTestSuite rewritten to Kotlin

### PR DESCRIPTION
**Pull request by PavelPunegov**

### CODE REVIEW

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    fun add(other: Statistics) {
+        passed += other.passed
+        failed += other.failed
+        error += other.error
+        skipped += other.skipped
+    }
+}
+
+open class KonanTestSuite(val name: String, val statistics: Statistics) {
```
Comment (olonho) [inline buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt]: abstract?

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+        open fun skip(): TestResult {
+            statistics.skip()
+            return TestResult(TestStatus.SKIPPED)
+        }
+    }
+
+    open fun createTestCase(name: String) = KonanTestCase(name)
+
+    open fun start() {}
```
Comment (olonho) [inline buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt]: Maybe skip body and require subclass to implement?

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt (hunk 1 above)
Comment (vvlevchenko) [reply buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt]: No, there `KonanTestSuite` used to be default implementation for command line launches and so on. The single child is for TeamCity execution.

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt (hunk 2 above)
Comment (PavelPunegov) [reply buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt]: By default KonanTestSuite is used by the standalone cli-exectution, while subclasses are used for the TeamCity

Comment (vvlevchenko) [reply buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/KonanTestSuite.kt]: I'd suggest to stay at this phase (naive and working groovy to kotlin conversion) to let me continue with refactoring and modification this part of code.

**Pull request merged**